### PR TITLE
Fix failing  mvd deployments

### PR DESCRIPTION
**Pull request by cpeeyush**
## What this PR changes/adds

_Briefly describe WHAT your pr changes, which features it adds/modifies._

## Why it does that

_Briefly state why the change was necessary._

## Further notes

_List other areas of code that have changed but are not necessarily linked to the main feature. This could be method
signature changes, package declarations, bugs that were encountered and were fixed inline, etc._

## Linked Issue(s)

Closes # <-- _insert Issue number if one exists_

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [ ] formatted title correctly?

### CODE REVIEW

**Review by chrislomonico — APPROVED** [review]

**Comment by chrislomonico** [comment]
draft looks good, as it is ready let me know and we will test it

File: deployment/curl-upgrade.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+curl --version
+sudo apt remove curl
```
Comment (chrislomonico) [inline deployment/curl-upgrade.sh:6]:
```suggestion
sudo apt remove -y curl
```

File: .github/workflows/deploy.yaml [hunk 2]
```diff
@@ -370,11 +378,19 @@ jobs:
       - name: 'Upload inbox storage account key as vault secret'
         run: az keyvault secret set --name "$INBOX_STORAGE_ACCOUNT-key1" --vault-name "$KEY_VAULT" --value "$INBOX_STORAGE_ACCOUNT_KEY" -o none
 
+      - name: 'Upgrade Curl'
+        run: |
+          chmod +x deployment/curl-upgrade.sh
```
Comment (algattik) [inline .github/workflows/deploy.yaml:383]:
```suggestion
```
commit the file with the right permissions

**Review by algattik — CHANGES_REQUESTED** [review]
would it be sufficient to run on agent `ubuntu-22.04`? includes curl 7.81.0, 7.71.0 is required.

https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2204-Readme.md

**Review by chrislomonico — CHANGES_REQUESTED** [review]
Per discussion, the curl update script needs some tweaking

**Pull request closed without merging**